sdk/core/azure-core/src/test/java/com/azure/android/core/internal/util/serializer/DateTimeSerializerTest.java

```diff
@@ -0,0 +1,183 @@
+
+            @Override
+            public JsonSerializer<Object> serializerInstance(Annotated annotated, Object serDef) throws JsonMappingException {
+                return null;
+            }
+
+            @Override
+            public Object includeFilterInstance(BeanPropertyDefinition forProperty, Class<?> filterClass) throws JsonMappingException {
+                return null;
+            }
+
+            @Override
+            public boolean includeFilterSuppressNulls(Object filter) throws JsonMappingException {
+                return false;
+            }
+        };
+        SerializerProvider serializerProviderStub =
+            new SerializerProviderStub(serializerProvider, serializationConfig, null);
+
+        // Actual serialization
+        new DateTimeSerializer().serialize(dateTime, jsonGenerator, serializerProviderStub);
+        jsonGenerator.flush();
+
+        assertEquals("1582592362000", outputStream.toString());
+    }
+
+    @Test
+    public void serializeDateTime_asString() throws IOException {
+        //DateTime
+        String testDate = "Tue, 25 Feb 2020 00:59:22 GMT";
+        OffsetDateTime dateTime =
+            OffsetDateTime.of(LocalDateTime.parse(testDate, RFC1123_DATE_TIME_FORMATTER), ZoneOffset.UTC);
+
+        // JsonGenerator and target OutputStream
+        StringBuilderOutputStream outputStream = new StringBuilderOutputStream();
+        JsonGenerator jsonGenerator = new JsonFactory().createGenerator(outputStream);
+
+        // SerializerProvider
+        BaseSettings baseSettings = new BaseSettings(null, new JacksonAnnotationIntrospector(), null,
+            TypeFactory.defaultInstance(), null, StdDateFormat.instance, null, Locale.getDefault(), null,
+            Base64Variants.getDefaultVariant());
+        SerializationConfig serializationConfig = new SerializationConfig(
+            baseSettings.withClassIntrospector(new BasicClassIntrospector()), new StdSubtypeResolver(),
+            new SimpleMixInResolver(null), new RootNameLookup(), new ConfigOverrides());
+        serializationConfig = serializationConfig.without(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+        SerializerProvider serializerProvider = new SerializerProvider() {
+            @Override
+            public WritableObjectId findObjectId(Object forPojo, ObjectIdGenerator<?> generatorType) {
+                return null;
+            }
+
+            @Override
+            public JsonSerializer<Object> serializerInstance(Annotated annotated, Object serDef) throws JsonMappingException {
+                return null;
+            }
+
+            @Override
+            public Object includeFilterInstance(BeanPropertyDefinition forProperty, Class<?> filterClass) throws JsonMappingException {
+                return null;
+            }
+
+            @Override
+            public boolean includeFilterSuppressNulls(Object filter) throws JsonMappingException {
+                return false;
+            }
+        };
```

Comment:
This appears to be duplicated

Reply:
I'll refactor the `SerializerProviderStub` class to include a constructor that creates an object like this, this way we avoid code duplication and make it easier to read what's going on.